### PR TITLE
[BugFix] Fix ignorance of where in AdminShowReplicaStatus

### LINKED ISSUE #32015: ADMIN SHOW REPLICA STATUS ignores WHERE clause
ADMIN SHOW REPLICA STATUS ignores WHERE clause. 
Based on https://docs.starrocks.io/en-us/latest/sql-reference/sql-statements/Administration/ADMIN%20SHOW%20REPLICA%20STATUS
I should be able to filter out non-OK  replicas but it is not working. I am always getting list of all replicas. 

### Steps to reproduce the behavior (Required)
ADMIN SHOW REPLICA STATUS FROM <db>.<table> WHERE STATUS = 'DEAD'

### Expected behavior (Required)
List of dead replicas or empty list if none exist.

### Real behavior (Required)
List of ALL replicas.

### StarRocks version (Required)
3.1

### PR DESCRIPTION
**Pull request by imay**
Fixes #32015

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [x] 3.1
  - [x] 3.0
  - [x] 2.5

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 1 / 1 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/AdminStmtAnalyzer.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

File: fe/fe-core/src/test/java/com/starrocks/sql/ast/AdminShowReplicaStatusStmtTest.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+import com.starrocks.catalog.Replica;
+import com.starrocks.qe.ConnectContext;
+import mockit.Mocked;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class AdminShowReplicaStatusStmtTest {
+    @Mocked
+    private ConnectContext ctx;
```
Comment (nshangyiming) [inline fe/fe-core/src/test/java/com/starrocks/sql/ast/AdminShowReplicaStatusStmtTest.java:25]:
```suggestion
    private ConnectContext ctx = UtFrameUtils.createDefaultCtx();
```

File: fe/fe-core/src/test/java/com/starrocks/sql/ast/AdminShowReplicaStatusStmtTest.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+    @Test
+    public void testNormal() throws Exception {
+        String sql = "ADMIN SHOW REPLICA STATUS FROM db1.tbl1 WHERE STATUS = 'DEAD'";
+        AdminShowReplicaStatusStmt stmt = (AdminShowReplicaStatusStmt) com.starrocks.sql.parser.SqlParser.parse(
+                sql, 32).get(0);
+        com.starrocks.sql.analyzer.Analyzer.analyze(stmt, ctx);
```
Comment (nshangyiming) [inline fe/fe-core/src/test/java/com/starrocks/sql/ast/AdminShowReplicaStatusStmtTest.java:32]:
```suggestion
        AdminShowReplicaStatusStmt stmt = (AdminShowReplicaStatusStmt) UtFrameUtils.parseStmtWithNewParser(showAlterStmtStr, ctx);
```

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 1 / 1 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/AdminStmtAnalyzer.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 1 / 1 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/AdminStmtAnalyzer.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 1 / 1 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/AdminStmtAnalyzer.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.2

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Pull request merged**